### ISSUE TITLE
[Staging] Using valid `prompt` parameter causes error in FxA

### ISSUE THREAD

**Issue by hidde** [mozilla/fxa-content-server#6316]
See also: 

Steps to reproduce: 
Trigger FxA with a URL that contains `prompt=select_account` (for example by follow the steps in [auth0-custom-lock#182](https://github.com/mozilla-iam/auth0-custom-lock/issues/182))

Expected result:
FxA login window is displayed.

Actual result: 
FxA displays an error:

<img width="535" alt="Fxa Error page: Bad Request: Invalid OAuth parameter: prompt" src="https://user-images.githubusercontent.com/178782/42023815-0d7a5476-7ac1-11e8-8c93-e9c691a509ea.png">

`select_account` is a valid value for the `prompt` parameter ([spec](https://openid.net/specs/openid-connect-core-1_0.html))

The problem only exists in the staging environment that is connected to Mozilla IAM, it works as expected in production.

**Comment by shane-tomlinson** [comment]
@hidde - we do not yet support this parameter, we only support `prompt=consent` at the moment.

Is your team looking to use both [prompt=none](https://github.com/mozilla/fxa-content-server/issues/6115) and prompt=select_account?

Changing the title of this bug to "support prompt=select_account"

**Comment by shane-tomlinson** [comment]
From the [OIDC document](https://openid.net/specs/openid-connect-core-1_0.html#AuthRequest):

> select_account
    The Authorization Server SHOULD prompt the End-User to select a user account. This enables an End-User who has multiple accounts at the Authorization Server to select amongst the multiple accounts that they might have current sessions for. If it cannot obtain an account selection choice made by the End-User, it MUST return an error, typically account_selection_required.

**Comment by hidde** [comment]
@shane-tomlinson Actually, not having the error and allowing user log in is all I'm after, not ‘real’  `select_account`support (although that would also be an interesting path to investigate), hope that helps.

Note that this is an issue that only happens in staging; in production Mozillians is using `prompt=select_account` and FxA works fine with that.

**Comment by hidde** [comment]
Yes. Seems like Auth0 is stripping it out, apologies I must have missed that when I verified it earlier. When the Auth0 login form (NLX) is opened it still has the parameter, but when Auth0's `/authorize` is called it is removed.

I guess we can fix this on our end then, should I close this issue?

**Comment by gdestuynder** [comment]
ideally it would be good it was supported by FxA:
we do want `prompt=none` so that auto-login works seamlessly, but if you start supporting `prompt` (which is `optional` in RFC2119 parlance), you'll want `prompt=select_account` even if we did not use it because it's a `should` on `prompt` is implemented.

**Comment by shane-tomlinson** [comment]
> we do want prompt=none so that auto-login works seamlessly, but if you start supporting prompt (which is optional in RFC2119 parlance)

I'd like to get https://github.com/mozilla/fxa-content-server/issues/6115 (prompt=none) prioritized for train-117, train-116 is already pretty heavily loaded. What is your timeline that you hoped to make use of that query parameter?

> you'll want prompt=select_account even if we did not use it because it's a should on prompt is implemented

Understood.

**Comment by gdestuynder** [comment]
I'd say the sooner the better - things currently work without but the user experience isnt as good as for other IdPs on Mozilla IAM right now (specially for prompt=none, though i'd implement everything the spec calls a "should" since it's the expectation for RPs)

**Comment by hmitsch** [comment]
Hi @shane-tomlinson,

> I'd like to get #6115 (prompt=none) prioritized for train-117

What's the approximate timeline for `train-117`? 

Depending on your answer we would hold off from building workarounds in NLX.

Best regards,
   Henrik

**Comment by shane-tomlinson** [comment]
Hi @hmitsch - unfortunately there's a lot more to `prompt=none` than I originally thought and isn't going to be on train-117, there's just too much involved. #5916 has the important context, we'll have to work through the issues outlined in https://github.com/mozilla/fxa-content-server/issues/5916#issuecomment-369777880 before we can add this feature.

**Comment by hmitsch** [comment]
Thanks for the update @shane-tomlinson. What's your current gut feeling about getting this into production? Is this not going to happen at all?

**Comment by shane-tomlinson** [comment]
> Thanks for the update @shane-tomlinson. What's your current gut feeling about getting this into production? Is this not going to happen at all?

I won't say no, but it's unlikely this quarter given our existing commitments.

**Comment by hidde** [comment]
For full transparency: in the meantime, we're [updating Mozilla IAM](https://github.com/mozilla/mozillians/pull/2528) to fix linking Firefox Accounts to profiles for now. We'll revert the fix whenever the `prompt=select_account` parameter stops yielding an error in FxA.

**Comment by hmitsch** [comment]
linking https://github.com/mozilla-iam/auth0-custom-lock/issues/182
